torch/distributed/_composable/checkpoint_activation.py

```diff
@@ -13,21 +13,22 @@
 
 
 @contextmanager
-def _no_hook(module: nn.Module):
+def _no_hook(module: nn.Module, user_ctx: ContextManager = nullcontext()):
```

Comment:
Note (mainly to myself): This `nullcontext()` object will be constructed at function definition time and reused for all invocations. We are assuming that we do not modify this object, so having it as a default argument is okay. (Otherwise, we would want `user_ctx  = None` and then inside the function, `user_ctx = nullcontext() if user_ctx is None else user_ctx`.)

Reply:
@awgu Although ``nullcontext()`` is initialized at function definition time. We initialize``_no_hook`` everytime in the pre-fw hook. So for any invocation if ``user_ctx is not None``, then ``user_ctx`` will be used.  

But I do see the safety is using what you suggested.